data_safe_haven/functions/strings.py

```diff
@@ -48,9 +48,11 @@ def next_occurrence(
             microsecond=0,
         )
         utc_dt = local_dt.astimezone(pytz.utc)
-        # Add one day until this datetime is in the future
-        utc_now = datetime.datetime.now(pytz.utc) + datetime.timedelta(minutes=1)
-        while utc_dt < utc_now:
+        # Add one day until this datetime is at least 1 hour in the future.
+        # This ensures that any Azure functions which depend on this datetime being in
+        # the future should treat it as valid.
+        utc_near_future = datetime.datetime.now(pytz.utc) + datetime.timedelta(hours=1)
+        while utc_dt < utc_near_future:
             utc_dt += datetime.timedelta(days=1)
```

Comment:
I think this is a lot clearer :+1:.

Reply:
Does the while loop ever cycle more than once?

Reply:
I think not, but dates/times are hard! I was taking a "better safe than sorry" approach here. Happy to change to an `if` if you think that's better?

Reply:
I know right :smile:.
I was trying to think of an edge case. Datetimes are always confusing. Let's leave it :+1:.